pkg/db/db_session/default.go

```diff
@@ -93,6 +94,44 @@ func (f *Default) Init(config *config.DatabaseConfig) {
 	})
 }
 
+func setPassword(dbConfig *config.DatabaseConfig) func(ctx context.Context, connConfig *pgx.ConnConfig) error {
+	return func(ctx context.Context, connConfig *pgx.ConnConfig) error {
+		if dbConfig.AuthMethod == constants.AuthMethodPassword {
+			connConfig.Password = dbConfig.Password
+			return nil
+		} else if dbConfig.AuthMethod == constants.AuthMethodMicrosoftEntra {
+			if isExpired(dbConfig.Token) {
+				token, err := getAccessToken(ctx, dbConfig)
+				if err != nil {
+					return err
+				}
+				connConfig.Password = token.Token
+				dbConfig.Token = token
+			}
```

Comment:
```suggestion
			if isExpired(dbConfig.Token) {
				token, err := getAccessToken(ctx, dbConfig)
				if err != nil {
					return err
				}
				connConfig.Password = token.Token
				dbConfig.Token = token
			} else {
			  connConfig.Password = dbConfig .Token.Token
			}
```

We need to handle the case where the token has not expired (or going to expire soon).

Reply:
Without this we might only be able to open one connection i.e one successfully `.Connect` call and all subsequent ones will fail with something like
```
2024-12-16T20:59:50.143Z	ERROR	logger/logger.go:129	  Unable to get all maestro instances: failed to connect to `host=<removed>.postgres.database.azure.com user=maestro-server database=maestro`: server error (FATAL: empty password returned by client (SQLSTATE 28P01)) 
github.com/openshift-online/maestro/pkg/logger.(*logger).log
````

Reply:
Yes. Good catch! Thanks.